sklearn/base.py

```diff
@@ -290,10 +290,15 @@ def __repr__(self):
                                                offset=len(class_name),),)
 
     def __getstate__(self):
+        try:
+            state = super(BaseEstimator, self).__getstate__()
+        except AttributeError:
+            state = self.__dict__
+
         if type(self).__module__.startswith('sklearn.'):
```

Comment:
I think we already test something like this. You can either test get/set_state directly (instead of pickling) or hack the pickle loading, perhaps by hacking `sys.modules`.